test-data/unit/pythoneval.test

```diff
@@ -1458,3 +1458,48 @@ _testNoCrashOnGenericUnionUnpacking.py:10: error: Revealed type is 'Union[builti
 _testNoCrashOnGenericUnionUnpacking.py:11: error: Revealed type is 'Union[builtins.str, builtins.int]'
 _testNoCrashOnGenericUnionUnpacking.py:15: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
 _testNoCrashOnGenericUnionUnpacking.py:16: error: Revealed type is 'Union[builtins.int*, builtins.str*]'
+
+[case testEnumIteration]
```

Comment:
I would use longer/more detailed names for test cases (there are lots of tests in `pythoneval.test`).

Reply:
Hm, not sure what more detail would be helpful here.

Reply:
Something like `EnumIterationPreservesPreciseType`

Reply:
That's only part of what this is aiming to test.  I'll add a comment in the test with more details.